### ISSUE TITLE
Crash when resizing while toggling float mode

### ISSUE THREAD

**Issue by tesselslate** [riverwm/river#847]
Happens on latest commit. Reproduction steps:

- Start with a window in tiled mode.
- Begin resizing the window with your cursor.
- Send the window back to tiled mode (with a keybind) and quickly begin resizing it again with your cursor.

Log:
[log.txt](https://github.com/riverwm/river/files/11793519/log.txt)

```
thread 40603 panic: reached unreachable code
/usr/lib/zig/std/debug.zig:278:14: 0x24f556 in assert (river)
    if (!ok) unreachable; // assertion failure
             ^
/home/dog/clones/river/river/Cursor.zig:980:73: 0x267c85 in updateState (river)
            assert(data.view.current.float or data.view.current.output.?.layout == null);
                                                                        ^
/home/dog/clones/river/river/Root.zig:677:74: 0x27a900 in commitTransaction (river)
        while (it) |node| : (it = node.next) node.data.cursor.updateState();
                                                                         ^
/home/dog/clones/river/river/Root.zig:589:31: 0x26bcc2 in notifyConfigured (river)
        self.commitTransaction();
                              ^
/home/dog/clones/river/river/XdgToplevel.zig:359:41: 0x26c0b2 in wrapper (river)
            server.root.notifyConfigured();
                                        ^
???:?:?: 0x7f5013e5901d in ??? (???)
```

**Comment by ifreund** [comment]
Hey, thanks for the bug report. I see the cause of this issue but there not a good direct fix yet.

I've made a bit of a mess of the move/resize code on master branch trying to handle mpv's not strictly protocol-compliant but mostly reasonable behavior and I introduced this bug in https://github.com/riverwm/river/pull/805

I've got an open PR (https://github.com/riverwm/river/pull/828) going even farther in the direction taken by #805 but I'm starting to thing it would be better to revert all of this new complexity and deal with mpv's weirdness in a different way.
